CMakeLists.txt

```diff
@@ -2,30 +2,67 @@ if (DEFINED CATKIN_DEVEL_PREFIX) # Detects if being built in a catkin context
     set(CATKIN ON)
 endif ()
 
+if ("$ENV{TARGET}" MATCHES "2021$")
+set(YEAR 2021)
+elseif ("ENV{TARGET}" MATCHES "2022$")
+set(YEAR 2022)
+else ()
+set(YEAR 2021) # default year to 2021
+endif ()
+
```

Comment:
I need to select which source files to build based on the year (putting 2021 files under `can/generated_2021` and 2022 files under `can/generated_2022`). This part works for firmware building but it wouldn't work on the ROS side, I'm not sure how to handle that yet